src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -207,5 +245,48 @@ public Task UnloadAsync()
 
             return Task.CompletedTask;
         }
+
+        // true means the project system plans to call NominateProjectAsync in the future.
+        public bool HasPendingNomination()
+        {
+            return _pendingNomination;
+        }
+
+        // NuGet calls this method to wait project to nominate restoring
+        // If the project has no pending restore data, it will return a completed task.
+        // Otherwise a task which will be completed once the project norminate the next restore
+        // the task will be cancelled, if the project system decide it no longer need restore (for example: the restore state has no change)
+        // the task will be failed, if the project system runs into a problem, so it cannot get correct data to norminate a restore (DT build failed)
+        public Task WhenNominated(CancellationToken cancellationToken)
+        {
+            if (_pendingNomination == false)
+            {
+                return Task.CompletedTask;
+            }
+            else
+            {
+                // task which will be completed once the project norminate the next restore (_norminateNextRestore == 1)
+                JoinableTask joinableTask = JoinableFactory.RunAsync(() =>
+                {
+                    while (true)
+                    {
+                        if (cancellationToken.IsCancellationRequested)
+                        {
+
+                        }
+
+                        if (_norminateNextRestore == true)
+                        {
+                            return Task.CompletedTask;
+                        }
+                    }
+                });
+
+                // Prevent overlap until Restore completes
+                return joinableTask.Task;
+            }
+        }
+
+        bool IVsProjectRestoreInfoSource.HasPendingNomination => _pendingNomination;
```

Comment:
Consider removing this member and changing the earlier `HasPendingNomination` method into a property.

Reply:
the method was removed